nengo_viz/viz.py

```diff
@@ -163,9 +166,25 @@ def load_config(self):
                 self.default_labels[v] = k
         return config
 
-    def save_config(self):
+    def save_config(self, lazy=False, force=False):
+        if not force and not self.config_save_needed:
+            return
+
+        now_time = time.time()
+        if not force and lazy and self.config_save_time is not None:
+            if (now_time - self.config_save_time) < self.config_save_period:
+                return
+
+        self.lock.acquire()
+        self.config_save_time = now_time
+        self.config_save_needed = False
         with open(self.filename + '.cfg', 'w') as f:
             f.write(self.config.dumps(uids=self.default_labels))
+        self.lock.release()
+
+    def modified_config(self):
+        if not self.config_save_needed:
```

Comment:
Probably better to just set it to true? That takes as long as the check to see if it's already true.